src/lib/validate-input.js

```diff
@@ -390,3 +390,23 @@ export function validateNotParams(input) {
     }
   }
 }
+
+export function validateGatherParams(input, indices, {axis = 0} = {}) {
+  if (axis !== undefined) {
+    const rank = input.rank;
+    if (!Number.isInteger(axis) || axis < 0) {
+      throw new Error(`The axis ${axis} should be an unsigned integer.`);
+    }
+    if (axis >= rank) {
+      throw new Error(`The axis ${axis} should be in the interval [0, ${rank}).`);
+    }
+  }
+  const axisSize = input.shape[axis];
+  for (let i = 0; i < sizeOfShape(indices.shape); ++i) {
+    const index = indices.getValueByIndex(i);
+    if (!Number.isInteger(index) || index < 0 || index >= axisSize) {
```

Comment:
The current WebNN DirectML backend allows negative index value. @fdwr

Reply:
https://www.w3.org/TR/webnn/#api-mlgraphbuilder-gather

Appears the spec settled on not supporting negative indices, `8. If index is greater than or equal to axisSize, then throw`, and that is tracked via https://github.com/webmachinelearning/webnn/issues/484.

I will look through the models I have for any negative indices to see if this will actually matter in practice...